src/avram/polymorphic.cr

```diff
@@ -0,0 +1,144 @@
+# Helpers for defining polymorphic associations
+module Avram::Polymorphic
+  # Sets up helper methods for validating and loading polymorphic associatiions
+  #
+  # This will generate methods for validiating, preloading, and lazy loading
+  # polymorphic associations.
+  #
+  # Let's start with this example:
+  #
+  # > *Important note:* the `belongs_to` associations must be nilable since only
+  # > one polymorphic can be set at a time. Remember to also make the association
+  # > nilable in the migration, for example: `add_belongs_to photo : Photo?`
```

Comment:
Does an exception get raised if you don't make it nillable? Is that even possible to catch at the time the polymorphic method is called?

Reply:
Not yet. It’s something that I’m trying to figure out how to check. You’ll probably notice it at runtime because you won’t be able to set just one polymorphic association without a database constraint error. I’d like to figure out how or if we can raise a better error. Either at compile time or at least at app start at runtime